docs/source/keylime/index.rst

```diff
@@ -0,0 +1,135 @@
+-----
+
+Build Keylime Server
+~~~~~~~~~~~~~~~~~~~~
+Provision the node:
+
+.. prompt:: bash $
+
+  metalsmith deploy --image <image> --network <network> --resource-class baremetal --ssh-public-key <path_to_public_key>
+
+* Note: the server needs to be connected to the public internet in order to download Keylime packages. Attaching the node to the public network could be as simple as running the command ``openstack esi node network attach (--network <network> | --port <port>) <node>``.
+
+Install Keylime packages:
+
+* After the node is active, ssh into it. There are various ways to acquire Keylime. One possible way is listed below:
+
+  .. prompt:: bash $
+
+    git clone https://github.com/keylime/keylime.git
+    cd keylime && ./installer.sh
+
+Modify /etc/keylime.conf:
+
+* Set ``require_ek_cert`` to false if the physical TPM does not have EK certificates.
+* Set the binding address for verifier and registrar, for example:
+
+  .. prompt::
+
+    [cloud_verifier]
+    cloudverifier_ip = 0.0.0.0
+
+    [registrar]
+    registrar_ip = 0.0.0.0
+
+Start Keylime services:
+
+.. prompt:: bash $
+
+  keylime_verifier
+  keylime_registrar
+
+Setup Keylime Agent
+~~~~~~~~~~~~~~~~~~~
+`Diskimage-builder tool`_ could be used to a build provisioning image with keylime-agent installed.
+
+* An example to use diskimage-builder tool to build a CentOS 8 image:
+
+  .. prompt:: bash $
+
+    export DIB_KEYLIME_AGENT_REGISTRAR_IP=<keylime_registrar_service_ip>
+    export DIB_KEYLIME_AGENT_REGISTRAR_PORT=<keylime_registrar_service_port>
+    export DIB_KEYLIME_AGENT_PORT=<keylime_agent_port> (default to 8890)
+    disk-image-create centos baremetal dhcp-all-interfaces grub2 keylime-agent -o keylime-image
+
+* TPM-emulator could be used instead of a real TPM hardware:
+
+  .. prompt:: bash $
+
+    disk-image-create centos baremetal dhcp-all-interfaces grub2 keylime-agent tpm-emulator -o keylime-image
+
+  Note: Keylime does not recommend TPM emulators for production systems and TPM 1.2 is no longer supported by Keylime.
+
+* Allowlist and its checksum could be extracted from the image initramfs or collected from the target node at boot time. Upload them to the machine that triggers attestation.
+
+* Upload the image to Glance service:
+
+  .. prompt:: bash $
+
+    KERNEL_ID=$(openstack image create \
+      --file keylime-image.vmlinuz --public \
+      --container-format aki --disk-format aki \
+      -f value -c id keylime-image.vmlinuz)
+    RAMDISK_ID=$(openstack image create \
+      --file keylime-image.initrd --public \
+      --container-format ari --disk-format ari \
+      -f value -c id keylime-image.initrd)
+    openstack image create \
+      --file keylime-image.qcow2 --public \
+      --container-format bare \
+      --disk-format qcow2 \
+      --property kernel_id=$KERNEL_ID \
+      --property ramdisk_id=$RAMDISK_ID \
+      keylime-image
+
+Alternatively, `Keylime ansible role`_ could be used to deploy Keylime against a hardware TPM.
```

Comment:
Does this go under 'Build Keylime Server'? Has it been tested with a leased node?

Reply:
This ansible role uses vagrant to create a VM with all keylime packages installed. The VM will be created inside the leased node. But that's not what people want. I guess I shouldn't mention this here.

Reply:
Yeah, if that's the case then I'd just drop it.

Reply:
Maybe I could put this link here for manual installation? https://github.com/keylime/keylime#manual